### PR TITLE
Contour2D/3D: eq (2)

### PR DESCRIPTION
**Pull request by sbendjebla**
* **Please check if the PR fulfills these requirements**
- [ ] The commit messages follows our guidelines (explicit, in english)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] CHANGELOG.md has been updated

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
- [x] Bug fix:
- [ ] New features:
- [ ] Documentation
- [ ] Optimization

* **Other information**:

It's the same version for **2D** & **3D**, maybe, we can put it in **Contour** ?!

### CODE REVIEW

File: volmdlr/wires.py [hunk 1]
```diff
@@ -1125,6 +1125,21 @@ def __init__(self, primitives: List[volmdlr.core.Primitive2D],
...
+            return False
+        equal = 0
+        for prim1 in self.primitives:
+            for prim2 in other_.primitives:
+                if (prim1 == prim2 or prim1.reverse() == prim2
+                        or prim2.reverse() == prim1 or prim1.reverse() == prim2.reverse()):
+                    equal += 1
+        if equal == len(self.primitives) and equal == len(other_.primitives):
+            return True
+        return False
```
Comment (AxelRinghausen) [inline volmdlr/wires.py:1141]: Just a small optimization.
```suggestion
    def __eq__(self, other_):
        if other_.__class__.__name__ != self.__class__.__name__:
            return False
        if len(self.primitives) != len(other_.primitives):
            return False
        equal = 0
        for prim1 in self.primitives:
            reverse1 = prim1.reverse()
            found = False
            for prim2 in other_.primitives:
                reverse2 = prim2.reverse()
                if (prim1 == prim2 or reverse1 == prim2
                        or reverse2 == prim1 or reverse1 == reverse2):
                    equal += 1
                    found = True
            if not found:
                return False
        if equal == len(self.primitives):
            return True
        return False
```

File: volmdlr/wires.py [hunk 2]
```diff
@@ -3159,12 +3174,19 @@ def __hash__(self):
...
```
Comment (AxelRinghausen) [inline volmdlr/wires.py:3189]: Small optimization.
```suggestion
        equal = 0
        for prim1 in self.primitives:
            reverse1 = prim1.reverse()
            found = False
            for prim2 in other_.primitives:
                reverse2 = prim2.reverse()
                if (prim1 == prim2 or reverse1 == prim2
                        or reverse2 == prim1 or reverse1 == reverse2):
                    equal += 1
                    found = True
            if not found:
                return False
        if equal == len(self.primitives):
            return True
        return False
```

**Review by AxelRinghausen — CHANGES_REQUESTED** [review]

File: volmdlr/wires.py (hunk 1 above)
Comment (sbendjebla) [reply volmdlr/wires.py]: Thanks !

File: volmdlr/wires.py (hunk 2 above)
Comment (sbendjebla) [reply volmdlr/wires.py]: Thanks !

**Pull request merged**